src/parquet/column/properties.h

```diff
@@ -0,0 +1,157 @@
+// under the License.
+
+#ifndef PARQUET_COLUMN_PROPERTIES_H
+#define PARQUET_COLUMN_PROPERTIES_H
+
+#include <memory>
+#include <string>
+
+#include "parquet/util/input.h"
+#include "parquet/util/mem-allocator.h"
+#include "parquet/types.h"
+
+namespace parquet {
+
+static int DEFAULT_PAGE_SIZE = 1024 * 1024;
+static int DEFAULT_DICTIONARY_PAGE_SIZE = DEFAULT_PAGE_SIZE;
+static bool DEFAULT_IS_DICTIONARY_ENABLED = true;
+
+class ReaderProperties {
+ public:
+  ReaderProperties(bool use_buffered_stream = false, int64_t buffer_size = 0,
+      MemoryAllocator* allocator = default_allocator()) {
+    enable_buffered_stream_ = use_buffered_stream;
+    buffer_size_ = buffer_size;
+    allocator_ = allocator;
+  }
+
+  static inline std::string type_printer(Type::type parquet_type, const char* val,
+      int length) {
+    std::stringstream result;
+    switch (parquet_type) {
+      case Type::BOOLEAN:
+        result << reinterpret_cast<const bool*>(val)[0];
+        break;
+      case Type::INT32:
+        result << reinterpret_cast<const int32_t*>(val)[0];
+        break;
+      case Type::INT64:
+        result << reinterpret_cast<const int64_t*>(val)[0];
+        break;
+      case Type::DOUBLE:
+        result << reinterpret_cast<const double*>(val)[0];
+        break;
+      case Type::FLOAT:
+        result << reinterpret_cast<const float*>(val)[0];
+        break;
+      case Type::INT96: {
+        for (int i = 0; i < 3; i++) {
+          result << reinterpret_cast<const int32_t*>(val)[i] << " ";
+        }
+        break;
+      }
+      case Type::BYTE_ARRAY: {
+        const ByteArray* a = reinterpret_cast<const ByteArray*>(val);
+        for (int i = 0; i < static_cast<int>(a->len); i++) {
+          result << a[0].ptr[i]  << " ";
+        }
+        break;
+      }
+      case Type::FIXED_LEN_BYTE_ARRAY: {
+        const FLBA* a = reinterpret_cast<const FLBA*>(val);
+        for (int i = 0; i < length; i++) {
+          result << a[0].ptr[i]  << " ";
+        }
+        break;
+      }
+      default:
+        break;
+    }
+    return result.str();
+  }
+
+  MemoryAllocator* allocator() {
+    return allocator_;
+  }
+
+  std::unique_ptr<InputStream> GetInputStream() {
+    std::unique_ptr<InputStream> stream;
+    if (enable_buffered_stream_) {
+      stream.reset(new BufferedInputStream(allocator_, buffer_size_));
+    } else {
+      stream.reset(new InMemoryInputStream());
+    }
+    return stream;
+  }
+
+  bool is_buffered_stream_enabled() {
+    return enable_buffered_stream_;
+  }
+
+  int64_t buffer_size() {
+    return buffer_size_;
+  }
+
+ private:
+  MemoryAllocator* allocator_;
+  bool enable_buffered_stream_;
+  int64_t buffer_size_;
+};
+
+ReaderProperties default_reader_properties();
+
+class WriterProperties {
+ public:
+  WriterProperties(int pagesize = DEFAULT_PAGE_SIZE,
+      int dictionary_pagesize = DEFAULT_DICTIONARY_PAGE_SIZE,
+      bool enable_dictionary = DEFAULT_IS_DICTIONARY_ENABLED,
```

Comment:
As the number of parameters grow, it may be better to have a ctor with minimal arguments and then add setter functions for each of the particular parameters.